### PR TITLE
Add new CredentialIdentifier selector with UI DSL support

### PR DESCRIPTION
**Pull request by abrooksv**
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt [hunk 1]
```diff
@@ -108,20 +107,27 @@ class ValidatingPanel internal constructor(
...
-    private inner class ValidatingAction(private val listener: (ActionEvent) -> Unit) : AbstractAction() {
-        override fun actionPerformed(e: ActionEvent) {
-            val errorList = performValidation()
-            if (errorList.isNotEmpty()) {
-                // Give the first error focus
-                val info = errorList.first()
-                info.component?.let {
-                    IdeFocusManager.getInstance(null).requestFocus(it, true)
-                }
+    fun hasValidInputs(): Boolean {
```
Comment (rli) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt:110]: method name doesn't convey that it has side effects, which is weird imo

Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt:110]: I can agree with that.

What do you think about `doValidation`?

Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/ValidatingPanel.kt]: run is fine

**Review by rli — APPROVED** [review]

File: jetbrains-core/src/software/aws/toolkits/jetbrains/ui/CredentialIdentifierSelector.kt [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+    private class InvalidCredentialIdentifier(override val id: String) : CredentialIdentifier {
+        override val displayName: String = message("credentials.invalid.not_found", id)
+        override val factoryId: String = "InvalidCredentialIdentifier"
+        override val credentialType: CredentialType? = null
+        override val defaultRegionId: String? = null
+    }
+
+    companion object {
+        fun Row.credentialSelector(
```
[the remark this answers is not in the document]
Comment (abrooksv) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/ui/CredentialIdentifierSelector.kt:92]: For now... will be used in future stuff.

**Pull request merged**